### PR TITLE
added *.development.config and *.development.json to gitignore

### PR DESCRIPTION
**Pull request by amsacha**
### Description
Allows users to store secrets and other sensitive info locally and not checking it in by mistake

### Contribution Checklist

- [x] The code builds clean without any errors or warnings
- [x] The PR follows SK Contribution Guidelines (https://github.com/microsoft/semantic-kernel/blob/main/CONTRIBUTING.md)
- [x] The code follows the .NET coding conventions (https://learn.microsoft.com/dotnet/csharp/fundamentals/coding-style/coding-conventions) verified with `dotnet format`
- [x] All unit tests pass, and I have added new tests where possible
- [x] I didn't break anyone :smile:

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -79,6 +79,10 @@ working with Memory, and see how the kernel works.
      [Polyglot extension](https://marketplace.visualstudio.com/items?itemName=ms-dotnettools.dotnet-interactive-vscode)
      to run the notebooks.
 
+### Working with Secrets
+
+We need keys to work with various aspects of the project including accessing openAI models. This opens up the possibility of exposing keys in commits. There are a [couple of options](https://learn.microsoft.com/en-us/aspnet/core/security/app-secrets?view=aspnetcore-7.0&tabs=windows) to safe guard developers from exposing keys. Outside of using the Secret Manager and environment variables, we've also added *.development.config to the .gitignore if developers want to use config files for secret storage.
```
Comment (adrianwyatt) [inline README.md:84]: should be `safeguard`. Same comment for the other READMEs in this PR.

File: README.md [hunk 2]
```diff
@@ -79,6 +79,10 @@ working with Memory, and see how the kernel works.
      [Polyglot extension](https://marketplace.visualstudio.com/items?itemName=ms-dotnettools.dotnet-interactive-vscode)
      to run the notebooks.
 
+### Working with Secrets
```
Comment (adrianwyatt) [inline README.md:82]: None of these projects use *.config files. I believe a better place for this comment would be in the KernelHttpServer README and perhaps just have a reference to that README in these files.

File: README.md (hunk 1 above)
Comment (adrianwyatt) [inline README.md:84]: Could you replace "Secret Manager" with "dotnet's user-secrets" and include a link to documentation? It's a relatively new feature that not everyone has used before.

**Review by adrianwyatt — CHANGES_REQUESTED** [review]

File: README.md (hunk 2 above)
Comment (amsacha) [reply README.md:82]: done

File: README.md (hunk 1 above)
Comment (adrianwyatt) [inline README.md:84]: Apologies - I didn't notice where this file was earlier. Let's keep this level of detail to the READMEs of the apps themselves and not in the top-level README.

File: samples/apps/auth-api-webapp-react/README.md [hunk 3]
```diff
@@ -27,6 +27,10 @@
    and/or `yarn start` from the command line.
 7. A browser will automatically open, otherwise you can navigate to `http://localhost:3000` to use the sample.
 
+### Working with Secrets
```
Comment (adrianwyatt) [inline samples/apps/auth-api-webapp-react/README.md:30]: [nit] Let's make this a quote, for example:

> **Working with secrets:** [KernelHttpServer's Readme](../dotnet/KernelHttpServer/README.md#Working-with-Secrets) has a note on safely working with keys and other secrets.

Comment (adrianwyatt) [reply samples/apps/auth-api-webapp-react/README.md:30]: Same for the other READMEs referencing KernelHttpServer

**Review by adrianwyatt — CHANGES_REQUESTED** [review]

**Review by adrianwyatt — APPROVED** [review]

**Pull request merged**